diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -50,17 +104,23 @@ class LineLength : Rule("line-length") {
         var offset = 0
         node.text.lines().forEach {
             if (it.length > configuration.lineLength) {
-                val newNode = node.psi.findElementAt(offset + it.trim().length - 1)!!.node
-                if ((newNode.elementType != KDOC_TEXT && newNode.elementType != KDOC_MARKDOWN_INLINE_LINK) || !isKDocValid(newNode)) {
+                val newNode = node.psi.findElementAt(offset + configuration.lineLength.toInt())!!.node
+                if ((newNode.elementType != KDOC_TEXT && newNode.elementType != KDOC_MARKDOWN_INLINE_LINK) ||
+                        !isKDocValid(newNode)) {
                     LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
                             "max line length ${configuration.lineLength}, but was ${it.length}",
-                            offset + node.startOffset) {}
+                            offset + node.startOffset, isFixable(newNode)) {
+                        positionByOffset = calculateLineColByOffset(node.treeParent.text)
+                        fixError(newNode, configuration)
+                    }
                 }
             }
             offset += it.length + 1
         }
     }
 
+    private fun isFixable(wrongNode: ASTNode) = wrongNode.parent({it.elementType in FIX_TYPE}) != null
```

Comment:
I believe almost every node in long line would have `FUN` among parents. This check seems not very reliable Maybe, you could create some common logic for this method and `fixError`?